cockpit/src/insights.jsx

```diff
@@ -0,0 +1,292 @@
+                                 style={{ "padding": "14px", "marginTop": "1ex", "marginBottom": "0px" }}>
+                              <p>{_("If you disconnect this system from Insights, it will no longer report it's Insights status in Customer Portal or Satellite.")}</p>
+                              <br/>
+                              <button className="btn btn-danger" onClick={left(disconnect)}>
+                                {_("Disconnect from Insights")}
+                              </button>
+                            </div>
+                          </Revealer>
+                        </div>
+                )
+            },
+            {
+                cancel_caption: _("Close"),
+                actions: [ ]
+            }
+        );
+
+        function disconnect() {
+            dlg.setFooterProps(
+                {
+                    cancel_caption: _("Cancel"),
+                    actions: [ ],
+                    idle_message: <div className="spinner spinner-sm"/>
+                });
+            unregister().then(
+                () => {
+                    dlg.footerProps.dialog_done();
+                },
+                error => {
+                    dlg.setFooterProps(
+                        {
+                            cancel_caption: _("Close"),
+                            actions: [ ],
+                            static_error: error.toString()
+                        });
+                })
+        }
+    }
+
+    get_lastupload_timestamp().then(lastupload => {
+        console.log(lastupload);
+        get_monotonic_start.then(monotonic_start => {
+            show(lastupload, monotonic_start);
+        });
+    });
+}
+
+
+export class InsightsStatus extends React.Component {
+    constructor() {
+        super();
+        this.state = { };
+        get_monotonic_start.then(start => this.setState({ monotonic_start: start }));
+        this.on_unit_changed = () => { this.setState({ }) };
+    }
+
+    componentDidMount() {
+        insights_timer.addEventListener("changed", this.on_unit_changed);
+        insights_service.addEventListener("changed", this.on_unit_changed);
+    }
+
+    componentWillUnmount() {
+        insights_timer.removeEventListener("changed", this.on_unit_changed);
+        insights_service.removeEventListener("changed", this.on_unit_changed);
+    }
+
+    render() {
+        let status;
+
+        if (!insights_timer.exists || !insights_service.exists)
+            return null;
+
+        if (insights_timer.enabled) {
+            status = (
+                    <div style={{display: "inline-block", verticalAlign: "top" }}>
+                    <a onClick={left(show_status_dialog)}>{_("Connected to Insights")}</a>
+                    { insights_service.state == "failed" && [ " ", <i className="pficon pficon-warning-triangle-o"/> ] }
+                    <br/>
+                    <a href="http://cloud.redhat.com" target="_blank" rel="noopener">
```

Comment:
I think linking directly to the Insights section on cloud.redhat.com makes more sense.
https://cloud.redhat.com/insights

Reply:
Done.

We should still try to get a link to the actual results page for this very machine, I'd say.